src/Controller/Admin/ThemeCatalogController.php

```diff
@@ -21,65 +21,76 @@
+     *     message="You do not have permission to update this.",
+     *     redirectRoute="admin_administration"
+     * )
      *
      * @return Response
      */
     public function indexAction(): Response
     {
-        $response = new Response();
+        $moduleUri = __PS_BASE_URI__ . 'modules/ps_mbo/';
+
+        $extraParams = [
+            'cdc_error_templating_url' => $moduleUri . 'views/js/cdc-error-templating.js',
+            'cdc_error_templating_css' => $moduleUri . 'views/css/cdc-error-templating.css',
+        ];
 
-        try {
-            $response->setContent($this->renderView(
-                '@Modules/ps_mbo/views/templates/admin/controllers/theme_catalog/addons_store.html.twig',
-                [
-                    'pageContent' => $this->get('mbo.externalcontent.provider')->getContent(
-                        $this->get('mbo.addons.links_provider')->getSelectionLink()
-                    ),
-                    'layoutHeaderToolbarBtn' => [],
-                    'layoutTitle' => $this->trans('Themes Catalog', 'Modules.Mbo.Themescatalog'),
-                    'requireAddonsSearch' => true,
-                    'requireBulkActions' => false,
-                    'showContentHeader' => true,
-                    'enableSidebar' => true,
-                    'help_link' => $this->generateSidebarLink($this->requestStack->getCurrentRequest()->get('_legacy_controller')),
-                    'requireFilterStatus' => false,
-                    'level' => $this->authorizationLevel($this->requestStack->getCurrentRequest()->get('_legacy_controller')),
-                ]
-            ));
-        } catch (ServiceUnavailableHttpException $exception) {
-            $response->setContent($this->renderView('@Modules/ps_mbo/views/templates/admin/error.html.twig'));
-            $response->setStatusCode($exception->getStatusCode());
-            $response->headers->add($exception->getHeaders());
+        $cdcJsFile = getenv('MBO_CDC_URL');
+        if (false === $cdcJsFile || !is_string($cdcJsFile) || empty($cdcJsFile)) {
+            $extraParams['cdc_script_not_found'] = true;
+            $extraParams['cdc_error_url'] = $moduleUri . 'views/js/cdc-error.js';
+        } else {
+            $extraParams['cdc_url'] = $cdcJsFile;
         }
 
-        return $response;
+
+        return $this->render(
+            '@Modules/ps_mbo/views/templates/admin/controllers/theme_catalog/addons_store.html.twig',
+            [
+                'layoutHeaderToolbarBtn' => $this->get('mbo.addons.toolbar')->getToolbarButtons(),
+                'layoutTitle' => $this->trans('Themes Catalog', 'Modules.Mbo.Themescatalog'),
+                'requireAddonsSearch' => true,
+                'requireBulkActions' => false,
+                'showContentHeader' => true,
+                'enableSidebar' => true,
+                'help_link' => $this->generateSidebarLink('AdminModules'),
```

Comment:
?

Reply:
Tu as 2 solutions pour avoir le request, soit tu fais function indexAction(Request $request)
Comme c'est un controller Symfony, Sf fera le reste

Ou tu fais $this->get('request_stack')->getCurrentRequest() ...